### PR TITLE
[docs] Write up doc explaining some basics about Android/iOS app size

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

Requested by Dan

# How

I took https://expo.fyi/android-app-size and expanded on it

# Preview

/distribution/app-size/

### CODE REVIEW

File: docs/pages/distribution/app-size.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+### Determining iOS app download and install size
+
+Typically, what app developers care about the "download size" on the App Store (what the users see in the store listing when they go to download the app). This will be the size of the split IPA that is generated by the store from your universal IPA.
+
+The only truly accurate way to see what your final app size will be shipped to users is to upload your app to the stores and download on a physical device. You can get accurate estimates from TestFlight: on [App Store Connect](https://appstoreconnect.apple.com/), navigate to TestFlight and select your build by clicking on the build number, then switch to the **Build Metadata** tab and click **App File Sizes**. You will see a list of estimated download and install sizes depending on the device type. Actually install sizes may also vary slightly depending on the iOS version on the device.
+
+## Android apps
+
+{/* todo: add info about minimum size of Android app on store */}
```
Comment (brentvatne) [inline docs/pages/distribution/app-size.mdx:41]: @aleqsio - did you end up looking into this at all, or just on the ios side? if not, could you? 🙏

File: docs/pages/distribution/app-size.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+title: Understanding app size
+description: Learn about how to determine what the actual size of your app will be when distributed to users, and how to get insights into your app size.
+---
+
+import { DocsLogo } from '@expo/styleguide';
+
+import { BoxLink } from '~/ui/components/BoxLink';
+import { Collapsible } from '~/ui/components/Collapsible';
+
+A common concern for developers is how much space their app takes up on the app store. This guide will help you understand what different build artifacts are for, figure out the actual size of your app will be when distributed to users, and learn how to get insights into and optimize your app size.
```
[the remark this answers is not in the document]
Comment (kadikraman) [reply docs/pages/distribution/app-size.mdx:11]: Also 
```diff
- how much space their app takes up on the app store
+ how much space their app takes up on the user's device
```

File: docs/pages/distribution/app-size.mdx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+**It probably isn't, actually!** When examining the resulting artifact of a release build for an app, it's common for developers who are unfamiliar with native Android and iOS development to be surprised by the file size — it's usually much larger than they would expect for an app of their scale if they were to download it from an app store. The reason is that **this is not actually the size of your app as it will be distributed on the app store!** When people talk about app sizes, what is relevant is the size of the app that they will download to their device, not the size that will be uploaded to app stores or shared in development and testing.
+
+There are various types of build artifacts that serve different purposes, and they are almost all larger than what users will see when they download your app from a store. This is because these builds are not optimized to target specific devices like they would when downloading from a store, but rather they typically include all of the code and resources that your app needs to run on a wide range of devices.
+
+## iOS apps
+
+The download size on the App Store of a minimal React Native app is just under 4 MB.
```
Comment (kadikraman) [inline docs/pages/distribution/app-size.mdx:21]:
```suggestion
The download size on the App Store of a minimal React Native app (created using the blank template) is just under 4 MB.
```
Do we link out to X from the docs? If so, maybe worth linking out to that tweet thread.

Comment (brentvatne) [reply docs/pages/distribution/app-size.mdx:21]: hm yeah i suppose we could do that!

File: docs/pages/distribution/app-size.mdx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+Of course, if you're running an incredibly popular app store that serves millions of users, you don't want to send the same 50 MB file to every single user, especially if they're only going to use a fraction of the resources in the APK. This is why the Google Play Store and other app stores have a feature called "App Bundles" (Android) that allows you to upload a single binary and then the store will generate a custom binary for each user based on their device's needs.
+
+### `.aab` (Android App Bundle)
+
+On Android, all new apps submitted to the Play Store must be built as an [Android App Bundle (.aab)](https://developer.android.com/platform/technology/app-bundle). Only once you have submitted the binaries to their respective stores will you be able to see the download size for various different device types.
```
[the remark this answers is not in the document]
Comment (brentvatne) [reply docs/pages/distribution/app-size.mdx:53]: nope

**Pull request merged**